pyaedt/edb_core/components.py

```diff
@@ -676,6 +676,60 @@ def _get_closest_pin_from(self, pin, ref_pinlist):
                 closest_pin = ref_pin
         return closest_pin
 
+    @pyaedt_function_handler()
+    def deactivate_rlc_component(self, component=None, create_circuit_port=False):
+        """Deactivate RLC component with possibility to replace them by circuit port.
+
+        Parameters
+        ----------
+        component : str
+            Component Refdes
+
+        create_circuit_port : bool
+            True, Component is converted into circuit port, False is just deactivated
+
+        Returns
+        -------
+        bool
+            True when succeeded, False when failed.
+
+        Examples
+        --------
+        >>> from pyaedt import Edb
+        >>> edb_folder = r'C:\my_edb_folder.aedb'
+        >>> edb = Edb(edb_folder)
+        >>> for cmp in list(edb.core_components.components.keys()):
+        >>>     edb.core_components.deactivate_rlc_component(component=cmp, create_circuit_port=False)
+        >>> edb.save_edb()
+        >>> edb.close_edb()
+        """
+        if not component:
+            return False
+        if isinstance(component, str):
+            component = self.components[component]
+            if not component:
+                self._logger.error("component {} not found.".format(component))
+                return False
+        if is_ironpython:
+            component_type = component.edbcomponent.GetComponentType()
+            if (
+                component_type == self._edb.Definition.ComponentType.Other
+                or component_type == self._edb.Definition.ComponentType.IC
+                or component_type == self._edb.Definition.ComponentType.IO
+            ):
+                self._logger.info("Component {} passed to deactivate is not an RLC.".format(component.refdes))
```

Comment:
For the logging it is recommended to use the old `%s` for string format. This prevent the message to be evaluated if the log is not required.
```suggestion
                self._logger.info("Component %s passed to deactivate is not an RLC.", component.refdes)
```